src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -323,6 +324,37 @@ public Mono<BasicStudyInfos> createStudy(Mono<FilePart> caseFile, String userId,
             });
     }
 
+    public Mono<BasicStudyInfos> createStudy(UUID sourceStudyUuid, UUID studyUuid, String userId) {
+        Objects.requireNonNull(sourceStudyUuid);
+
+        AtomicReference<Long> startTime = new AtomicReference<>();
+        UUID newNetworkUuid = UUID.randomUUID();
+        UUID reportUuid = UUID.randomUUID();
+
+        StudyEntity sourceStudy = doGetStudy(sourceStudyUuid);
+        RootNode rootNode = networkModificationTreeService.doGetStudyTree(sourceStudyUuid);
+
+        return insertStudyCreationRequest(userId, studyUuid)
+                .doOnSubscribe(x -> startTime.set(System.nanoTime()))
+                .map(StudyService::toBasicStudyInfos)
+                .doOnSuccess(ns -> {
```

Comment:
ns ?

Reply:
It stands for new study, i've changed it to newStudyCreationRequest